### PR TITLE
delay the check if the remote is an owncloud instance till we actually use the scanner

### PR DESCRIPTION
**Pull request by icewind1991**
Fixes the performance regression introduced with 9ca1e9c7e941ee04a33bdf092739df367fe68356

cc @PVince81 @LukasReschke

### CODE REVIEW

File: apps/files_sharing/lib/external/storage.php [hunk 1]
```diff
@@ -253,11 +250,15 @@ private function testRemoteUrl($url) {
 	 *
 	 * @return bool
 	 */
-	private function remoteIsOwnCloud() {
-		if(defined('PHPUNIT_RUN') || !$this->testRemoteUrl($this->getRemote() . '/status.php')) {
-			return false;
+	public function remoteIsOwnCloud() {
+		static $isOwncloud = null;
```
Comment (PVince81) [inline apps/files_sharing/lib/external/storage.php]: please stop using local `static` and declare this as a local attr

File: apps/files_sharing/lib/external/scanner.php [hunk 2]
```diff
@@ -54,7 +58,7 @@ public function scan($path, $recursive = self::SCAN_RECURSIVE, $reuse = -1, $loc
 	 */
 	public function scanFile($file, $reuseExisting = 0, $parentId = -1, $cacheData = null, $lock = true) {
 		try {
-			return parent::scanFile($file, $reuseExisting);
+			return parent::scanFile($file, $reuseExisting, $parentId, $cacheData, $lock);
```
Comment (PVince81) [inline apps/files_sharing/lib/external/scanner.php]: Is this part of the fix ?

File: apps/files_sharing/lib/external/storage.php (hunk 1 above)
Comment (icewind1991) [inline apps/files_sharing/lib/external/storage.php]: done

Comment (LukasReschke) [inline apps/files_sharing/lib/external/storage.php]: We could also use the memory cache here and store the results.

File: apps/files_sharing/lib/external/scanner.php (hunk 2 above)
Comment (icewind1991) [inline apps/files_sharing/lib/external/scanner.php]: yes

Comment (icewind1991) [inline apps/files_sharing/lib/external/scanner.php]: making sure that it behaves the same as the parent method

**Comment by PVince81** [comment]
Tested against another OC, works :+1: 
I also checked that whenever the other OC becomes unavailable (maintenance mode) then back again, PROPFIND still works.
Also, adding a file directly on the remote gets properly detected by the scanner.

**Comment by LukasReschke** [comment]
I will test locally whether this still works with Pydio :hourglass_flowing_sand:

**Comment by LukasReschke** [comment]
This breaks sharing stuff with Pydio. I'll adjust this locally to get it work…

**Comment by LukasReschke** [comment]
My approach: https://github.com/owncloud/core/pull/22859

**Comment by karlitschek** [comment]
@LukasReschke @icewind1991 @DeepDiver1975 @PVince81 What do we do with this?

**Comment by LukasReschke** [comment]
> @LukasReschke @icewind1991 @DeepDiver1975 @PVince81 What do we do with this?

Personally I don't think this is absolutely release critical. While it is a performance penalty it is from my understanding not something that would break stuff. But it may indeed annoy people so your call whether we want to get this into 9.0.0 or 9.0.1.

However, I can easily open a backport PR of https://github.com/owncloud/core/pull/22859. That one is rather low risk. It's not a perfect solution but IMHO good enough (at least if you have a local cache configured). The backport can be found at https://github.com/owncloud/core/pull/22884

**Comment by karlitschek** [comment]
It seems that we have to redo the packages anyhow. So let's do this for 9.0.0

**Comment by cmonteroluque** [comment]
ok

**Comment by MorrisJobke** [comment]
#22859 replaced this one -> @LukasReschke So we could close it right? (please reopen if I'm wrong here)

**Pull request closed without merging**